API.md

```diff
@@ -90,3 +90,65 @@ _Note: The `message` field is meant only to describe the error to a user, this t
 
 `{"error":{"id":"no_login","message":"You are not logged in to an account"},"status":"error"}`
 
+### `/resetpass` - Reset a password for a user
+#### Params
+- `account`
+- `email`
+
+#### Responses
+##### Success
+
+`{"status":"ok"}`
+
+#### Errors
+
+`{"error":{"id":"no_account","message":"Unable to find matching account"},"status":"error"}`
+
+`{"error":{"id":"mail_failed","message":"Unable to send reset email"},"status":"error"}`
+
+### `/resetpass/confirm` - Confirm a password reset
+This has two main uses:
+- Simply confirming the reset request and logging in the user, which will then mark the account as unconfirmed until the user changes their password, or
+- Confirming the reset request and also setting the password to `newpass` in one request (requires sending the `newpass` parameter
+
+This allows finer control from the API consumer side as to the separation of authentication and modification of the account
+
+#### Params
+- `account` - Account name to reset the password for
+- `code` - Confirmation code from the user
+- `newpass` _optional_ - If set, sets the new password in this request, otherwise a later request to `/user/set/password` is required.
+
+#### Responses
+##### Success
+
+`{"status":"ok","password_set":true|false,"account":"<ACCOUNT_NAME>","verified":true|false,"session":"<SESSION_ID>"}`
+
+- `password_set` - Was the password changed in this request or is a future request to `/user/set/password` required
+- `verified` - Is the account a verified account or not
+
+##### Errors
+
+`{"error":{"id":"no_account","message":"Unable to find matching account"},"status":"error"}`
```

Comment:
`no_account` and `no_error` should simply be `error` with "unable to complete request". Expired code is fine as it already verified that the user has the code

Reply:
What is `no_error` ?

Reply:
sorry, I mistyped, I wanted `wrong_code`

Reply:
ah yeah, I agree, will merge those

Reply:
I could simply merge no_account to wrong_code since technically the code is invalid for a non-existent account as well

Reply:
That works

Reply:
Done in 1fd4bdfd112a337eb759f82cf4bf24e18d373acb